src/resolve/adapters/baseSourceAdapter.ts

```diff
@@ -146,6 +146,24 @@ export abstract class BaseSourceAdapter implements SourceAdapter {
     }
   }
 
+  private calculateName(rootMetadata: MetadataXml): string {
+    if (!this.type.folderType) {
+      return rootMetadata.fullName;
+    }
+    const grandparentType = this.registry.getTypeByName(this.type.folderType);
+    if (grandparentType.folderType && grandparentType.folderType !== this.type.id) {
+      const splits = rootMetadata.path.split(sep);
```

Comment:
Using path sep can be tricky because if the path is from a server response it will necessarily be a forward slash, even on Windows OS.  If this is used for a retrieve on Windows I'd expect it to not work as expected.  In order for it to work we first have to normalize the path, then split.